### ISSUE TITLE
client.vx.list_stock_financials() appears to be ignoring date range operators

### ISSUE THREAD

**Issue by bfoz** [polygon-io/client-python#382]
**Describe the bug**

Fetching financial reports for a "period of report date" range returns more than the reports for the given range (a lot more). 

**To Reproduce**

```
reports = client.vx.list_stock_financials(ticker="AAPL", timeframe='annual', period_of_report_date_gte: datetime.date(2023, 1, 1), period_of_report_date_lt: datetime.date(2024, 1, 1))
for report in reports:
    print(f"{report.fiscal_year=}, {report.fiscal_period=}")
```

I've also tried passing the dates as strings instead of date() objects, but that didn't change the result.

**Expected behavior**

The API should only return reports for the given date range. Instead, it returns...

```
report.fiscal_year='2022', report.fiscal_period='FY'
report.fiscal_year='2021', report.fiscal_period='FY'
report.fiscal_year='2020', report.fiscal_period='FY'
report.fiscal_year='2019', report.fiscal_period='FY'
report.fiscal_year='2018', report.fiscal_period='FY'
report.fiscal_year='2017', report.fiscal_period='FY'
report.fiscal_year='2016', report.fiscal_period='FY'
report.fiscal_year='2015', report.fiscal_period='FY'
report.fiscal_year='2014', report.fiscal_period='FY'
report.fiscal_year='2013', report.fiscal_period='FY'
report.fiscal_year='2012', report.fiscal_period='FY'
report.fiscal_year='2011', report.fiscal_period='FY'
report.fiscal_year='2010', report.fiscal_period='FY'
report.fiscal_year='', report.fiscal_period=''
```

**Comment by jakekdodd** [comment]
The issue is [here](https://github.com/polygon-io/client-python/blob/4d8ab1c45da67a036e1460b1a25cb5158598a846/polygon/rest/base.py#L151):
```python
if (
    argname.endswith("_lt")
    or argname.endswith("_lte")
    or argname.endswith("_gt")
    or argname.endswith("_gte")
    or argname.endswith("_any_of")
):
    argname = ".".join(argname.split("_", 1))
```

This is supposed to convert (for example) the python method param `period_of_report_date_gte` to the query param `period_of_report_date.gte` for the API request. Instead, it's converting it to `period.of_report_date_gte`. This is a problem affecting all API methods, for any parameter that uses additional filter params. 

Note that you can currently get around this bug by using the `params=` keyword arg instead. So in your case,
```python
reports = client.vx.list_stock_financials(
    ticker="AAPL", 
    timeframe='annual', 
    params={
        "period_of_report_date.gte": str(datetime.date(2023, 1, 1)),
        "period_of_report_date.lt": str(datetime.date(2024, 1, 1))),
    },
)
```

Happy to submit a PR...I suggest simply changing that block to something equivalent to:
```python
for ext in ["lt", "lte", "gt", "gte", "any_of"]:
    if argname.endswith(f"_{ext}"):
        argname = argname.removesuffix(f"_{ext}") + f".{ext}")
```

**Comment by justinpolygon** [comment]
FYI - We just merged the fix and rolled a new release. So, this should be fixed now via polygon-api-client 1.7.2.
